### PR TITLE
Feature/auth and posts

### PR DESCRIPTION
**Pull request by ambid17**
Added register/login, creating/viewing posts, and an account page

### CODE REVIEW

File: Sources/App/Controllers/API/PostController.swift [hunk 1]
```diff
@@ -18,12 +18,12 @@ class PostController: RouteCollection {
     func boot(router: Router) throws {
         let authSessionRouter = router.grouped(User.authSessionsMiddleware())
         
-        authSessionRouter.get("api", "post", PostItem.parameter, use: postHandler)
+        authSessionRouter.get("api", "post", Int.parameter, use: postHandler)
```
Comment (0xLeif) [inline Sources/App/Controllers/API/PostController.swift]: Revert all `Int.parameter`

File: Sources/App/Controllers/API/PostController.swift [hunk 2]
```diff
@@ -38,7 +38,7 @@ class PostController: RouteCollection {
                         if let _ = result {
                             return req.future(error: BadPost())
                         }
-                        
+                        // req.isClient == Browser
```
Comment (0xLeif) [inline Sources/App/Controllers/API/PostController.swift]: Remove

File: Sources/App/Controllers/API/UserController.swift [hunk 3]
```diff
@@ -65,9 +62,9 @@ class UserController: RouteCollection {
                 on: req
             ).map { user in
                 guard let user = user else {
+                    print("bad")
```
Comment (0xLeif) [inline Sources/App/Controllers/API/UserController.swift]: bad

File: Sources/App/Controllers/AuthRouteController.swift [hunk 4]
```diff
@@ -7,24 +7,106 @@ class AuthRouteController: RouteCollection {
     func boot(router: Router) throws {
         let authSessionRouter = router.grouped(User.authSessionsMiddleware())
         let protectedRouter = authSessionRouter.grouped(RedirectMiddleware<User>(path: "/login"))
-        protectedRouter.get(use: indexHandler)
+        protectedRouter.get("dashboard", use: dashboardHandler)
+        protectedRouter.get("authPosts", use: listPostsHandler)
+        protectedRouter.get("authIndex", use: authIndexHandler)
+        protectedRouter.get("createPost", use: createPostHandler)
+        protectedRouter.get("viewPost", PostItem.parameter, use: viewPostHandler)
```
Comment (0xLeif) [inline Sources/App/Controllers/AuthRouteController.swift:14]: PostItem.parameter > Int.parameter

**Review by 0xLeif — CHANGES_REQUESTED** [review]

**Review by will-nemo — APPROVED** [review]

**Review by 0xLeif — APPROVED** [review]

**Pull request merged**